### ISSUE TITLE
NGINX 502 error with MongoDB and Mongo Express

### ISSUE THREAD

**Issue by tim-rohrer** [caprover/one-click-apps#348]
I created MongoDB and Mongo Express instances from the one click options. I thought it worked at first, but now I get the NGINX 502 error whenever I try to view a database.

Here is what I've done, checked or not done:
1. Confirmed both apps are running. I am able to bring up the ME screen and see the three default databases. However, when I click on "view", I get the 502 error.
2. `docker volume ls` shows both persistent volumes.
3. The log for Mongo shows, if I understand it, authentication against the admin database:
``` bash
{"t":{"$date":"2021-01-16T01:00:51.147+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"10.0.1.11:54460","connectionId":22,"connectionCount":3}}
{"t":{"$date":"2021-01-16T01:00:51.253+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn22","msg":"Successful authentication","attr":{"mechanism":"SCRAM-SHA-1","principalName":"root","authenticationDatabase":"admin","client":"10.0.1.11:54460"}}
{"t":{"$date":"2021-01-16T01:00:51.261+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn22","msg":"Connection ended","attr":{"remote":"10.0.1.11:54460","connectionId":22,"connectionCount":2}}
{"t":{"$date":"2021-01-16T01:00:51.262+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn20","msg":"Connection ended","attr":{"remote":"10.0.1.11:53852","connectionId":20,"connectionCount":1}}
{"t":{"$date":"2021-01-16T01:00:51.263+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn21","msg":"Connection ended","attr":{"remote":"10.0.1.11:53854","connectionId":21,"connectionCount":0}}
```
4. The ME log shows connection:

```bash
Waiting for srv-captain--mongodb:27017...
Welcome to mongo-express
------------------------

Mongo Express server listening at http://0.0.0.0:8081
Server is open to allow connections from anyone (0.0.0.0)
Database connected
Admin Database connected
```
5. Restarted the host VPS.
6. I haven't changed any of the App Config settings for either service except to add (as a troubleshooting step) HTTPS for ME. However, I am not forcing it.
7. Read the [docs on this issue](https://caprover.com/docs/troubleshooting.html#successful-deploy-but-502-bad-gateway-error)
8. Searched both caprover and this repo for matching issues or additional suggestions.
9. `ufw disable`
10. Created a new database (successfully) but still get the error if I try to view it. And I'm confused by the namespace???

```bash
{"t":{"$date":"2021-01-16T01:31:05.925+00:00"},"s":"I",  "c":"STORAGE",  "id":20320,   "ctx":"conn35","msg":"createCollection","attr":{"namespace":"Test.delete_me","uuidDisposition":"generated","uuid":{"uuid":{"$uuid":"54452b55-f01d-400a-8121-a681d28cb589"}},"options":{}}}
{"t":{"$date":"2021-01-16T01:31:05.932+00:00"},"s":"I",  "c":"INDEX",    "id":20345,   "ctx":"conn35","msg":"Index build: done building","attr":{"buildUUID":null,"namespace":"Test.delete_me","index":"_id_","commitTimestamp":{"$timestamp":{"t":0,"i":0}}}}
```

I apologize if I've missed something, but I would appreciate some additional steps I can take.

Thank you.

**Comment by githubsaturn** [comment]
It looks like a MongoExpress bug. Is this the entire log from MongoExpress? Even after it displays a 502?
```
Waiting for srv-captain--mongodb:27017...
Welcome to mongo-express
------------------------

Mongo Express server listening at http://0.0.0.0:8081
Server is open to allow connections from anyone (0.0.0.0)
Database connected
Admin Database connected
```

**Comment by githubsaturn** [comment]
Yes, looks like it's a mongo express bug. See here: https://github.com/mongo-express/mongo-express/issues/459

**Comment by githubsaturn** [comment]
More issues: https://github.com/mongo-express/mongo-express/search?q=Cannot+read+property+%27toString%27+of+undefined&type=issues

**Comment by githubsaturn** [comment]
Looks like it may have been fixed in `mongo-express:0.54` image. Do this:
- Go to Apps > Mongo Express
- Go to Deployment tab
- From "**Method 6: Deploy via ImageName**" deploy this image: `mongo-express:0.54`

**Comment by tim-rohrer** [comment]
I'll research that. Interesting that I don't have that issue on my ME locally installed with Docker (I believe using the latest image).

Does the one click app pull the latest images or a set image?

**Comment by githubsaturn** [comment]
Whatever image you deploy, it'll remain the same until you manually deploy another image on top of it.

**Comment by tim-rohrer** [comment]
> Looks like it may have been fixed in `mongo-express:0.54` image. Do this:
> 
> * Go to Apps > Mongo Express
> * Go to Deployment tab
> * From "**Method 6: Deploy via ImageName**" deploy this image: `mongo-express:0.54`

Perfect. That works. Thank you. 

Good catch on the ME bug. I didn't even think that might be it since I wasn't having troubles my own recently set up ME (within the past few days).

Have a good night, and I do appreciate your quick response.
